tests/test_internals/test_movie_interface.py

```diff
@@ -0,0 +1,143 @@
+from itertools import product
+
+import numpy as np
+import pytest
+from pynwb import NWBHDF5IO
+
+from nwb_conversion_tools import NWBConverter, MovieInterface
+
+try:
+    import cv2
+
+    HAVE_OPENCV = True
+except ImportError:
+    HAVE_OPENCV = False
+
+
+@pytest.fixture(scope="module")
+def create_movies(tmp_path_factory):
+    if HAVE_OPENCV:
```

Comment:
Could you implement this testing refactor through `unittest.TestCase`? See https://github.com/catalystneuro/nwb-conversion-tools/issues/287

Using common `setUp` techniques could simplify a lot of the logic, and it's always good to have separate tests for each option/use-case as you are starting to separate these

Reply:
so do we plan to use Unittest all across as a convention? 
The fixtures behave like the `setUp` and with the `yield` statement, they also have a teardown functionality by running the code after yield during cleanup. I have appropriately kept some fixtures with module level scope and others with function scope (default for `purest.fixture`). 
I find `pytest` easy to use, is there a reason why you think `unittest` is better?
I have made some changes from your last comment, please review: https://github.com/catalystneuro/nwb-conversion-tools/pull/372/commits/64cf4639f0173ba82837eb26a6bfe1c22cf43073

Reply:
Convention, yes, requirement, no.

Overall it offers superior code organization since you can group similar tests (data requirements, behavioral methods, etc.) into similar classes all in the same file. Also `hdmf.testing TestCase` (child class of `unittest`) already has a ton of added useful functionality for various things, such as testing exact assertions. It's also easier for new users to understand, since a `yield` -> clean-up approach actually gets pretty complicated, as does defining multiple helper functions across the scope for things that would otherwise be simple attributes of the class.

I grant you, the original tests for this interface were not well organized either way so anything can be done here to improve them; but in the future if we do attempt a refactor with `unittest`, I'd swap to it if it offers clearer organization or code reduction.